lib/resin.coffee

```diff
@@ -219,3 +224,64 @@ module.exports = getSdk = (opts = {}) ->
 	sdk.errors = errors
 
 	return sdk
+
+###*
+# @summary Set shared default options
+# @name setSharedOptions
+# @public
+# @function
+# @memberof resin
+#
+# @description
+# Set options that are used by calls to `resin.fromSharedOptions()`.
+# The options accepted are the same as those used in the main SDK factory function.
+#
+# @params {Object} opts - The shared default options
+#
+# @example
+# resin.setSharedOptions({
+# 	apiUrl: 'https://api.resin.io/',
+# 	imageMakerUrl: 'https://img.resin.io/',
+# 	apiVersion: 'v2',
+# 	isBrowser: true,
+# });
+###
+getSdk.setSharedOptions = (opts) ->
+	defaults(opts, defaultOpts)
+
+	root = window || GLOBAL
+
+	if root[RESIN_SDK_HAS_USED_SHARED_OPTIONS]
+		console.error('Shared SDK options have already been used. You may have a race condition in your code.')
+
```

Comment:
I think we should handle the case where you accidentally set the shared options twice too, since that's definitely a bug.